extensions/guacamole-auth-wol/src/main/resources/services/wolService.js

```diff
@@ -0,0 +1,158 @@
+            data    : $.param({connectionIdentifier: identifier}),
+            headers : {'Content-Type': 'application/x-www-form-urlencoded'}
+        })
+        .then(function wokeHost(wokeData) {
+            return wokeData['identifier'];
+        });
+    };
+    
+    /**
+     * Retrieve host information about the host associated with the given
+     * connection data source and identifier.
+     * 
+     * @param {type} dataSource
+     *     The data source where the connection exists.
+     *     
+     * @param {type} identifier
+     *     The identifier of the connection.
+     *     
+     * @returns {Promise}.Object
+     *     A promise, which, when resolved, will contain the hostname and
+     *     port associated with the connection.
+     */
+    var getHostInfo = function getHostInfo(dataSource, identifier) {
+        
+        return connectionService.getConnectionParameters(dataSource, identifier)
+        .then(function gotParameters(paramData) {
+            
+                $log.debug(paramData);
+            
+                return {
+                   'hostname' : paramData['hostname'],
+                   'port'     : paramData['port']
+               };
+                            
+        }, function(error) {
+            $log.error(error);
+            return null;
+        });
+    };
+    
+    /**
+     * Check the status of the host associated with the given datasource
+     * and connection identifier, returning a promise which, when resolved,
+     * will contain true if the host is up, or false if the host is down or an
+     * error occurs.
+     * 
+     * @param {type} dataSource
+     *     The datasource in Guacamole where the connection exists.
+     *     
+     * @param {type} identifier
+     *     The identifier of the connection.
+     *     
+     * @return {Promise}.Boolean
+     *     A promse which, when resolved, will contain a Boolean value for the
+     *     status of the host - True if the host is up, false otherwise or if
+     *     an error exists.
+     */
+    service.checkStatus = function checkStatus(dataSource, identifier) {
+
+        // Build HTTP parameters set
+        var httpParameters = {
+            token : authenticationService.getCurrentToken()
+        };
+        
+        // Retrieve the information for the host (hostname and port).
+        var hostInfo = getHostInfo(dataSource, identifier);
+        
+        // Retrieve a promise to get the status of the host.
+        var status = hostInfo.then(function(hostData) {
+        
+            // Send the WOL request
+            return requestService({
+                method  : 'GET',
+                url     : 'api/session/ext/wol/check/' + hostData['hostname'] + '/' + hostData['port'],
+                params  : httpParameters
+            })
+            .then(function gotStatus(statusData) {
+                return statusData;
+            }, function(error) {
+                $log.error(error);
```

Comment:
If a warning is sufficient, you may be able to leverage `requestService.WARN`:

https://github.com/apache/guacamole-client/blob/3089e71e60541731a49c8d8a5d69a7425780ce6f/guacamole/src/main/webapp/app/rest/services/requestService.js#L133-L144